src/unix/linux_like/android/b32/arm.rs

```diff
@@ -1,5 +1,167 @@
 pub type c_char = u8;
 pub type wchar_t = u32;
+pub type greg_t = i32;
+pub type mcontext_t = sigcontext;
+
+s! {
+    pub struct sigcontext {
+        pub trap_no: ::c_ulong,
+        pub error_code: ::c_ulong,
+        pub oldmask: ::c_ulong,
+        pub arm_r0: ::c_ulong,
+        pub arm_r1: ::c_ulong,
+        pub arm_r2: ::c_ulong,
+        pub arm_r3: ::c_ulong,
+        pub arm_r4: ::c_ulong,
+        pub arm_r5: ::c_ulong,
+        pub arm_r6: ::c_ulong,
+        pub arm_r7: ::c_ulong,
+        pub arm_r8: ::c_ulong,
+        pub arm_r9: ::c_ulong,
+        pub arm_r10: ::c_ulong,
+        pub arm_fp: ::c_ulong,
+        pub arm_ip: ::c_ulong,
+        pub arm_sp: ::c_ulong,
+        pub arm_lr: ::c_ulong,
+        pub arm_pc: ::c_ulong,
+        pub arm_cpsr: ::c_ulong,
+        pub fault_address: ::c_ulong,
+    }
+}
+
+cfg_if! {
+    if #[cfg(libc_union)] {
+        s_no_extra_traits! {
+            pub struct __c_anonymous_uc_sigmask_with_padding {
+                pub uc_sigmask: ::sigset_t,
+                /* Android has a wrong (smaller) sigset_t on x86. */
+                __padding_rt_sigset: u32,
+            }
+
+            pub union __c_anonymous_uc_sigmask {
+                uc_sigmask: __c_anonymous_uc_sigmask_with_padding,
+                uc_sigmask64: ::sigset64_t,
+            }
+
+            pub struct ucontext_t {
+                pub uc_flags: ::c_ulong,
+                pub uc_link: *mut ucontext_t,
+                pub uc_stack: ::stack_t,
+                pub uc_mcontext: mcontext_t,
+                pub uc_sigmask__c_anonymous_union: __c_anonymous_uc_sigmask,
+                /* The kernel adds extra padding after uc_sigmask to match
+                 * glibc sigset_t on ARM. */
+                __padding: [c_char; 120],
+                __align: [::c_longlong; 0],
```

Comment:
Oh, I forgot to ask about alignment.
In `ucontext_t` for Android ARM32, `uc_regspace` is defined as follows:
(from https://android.googlesource.com/platform/bionic/+/refs/tags/ndk-r20/libc/include/sys/ucontext.h#100)

```c
typedef struct ucontext {
  ...
  unsigned long uc_regspace[128] __attribute__((__aligned__(8)));
} ucontext_t;
```

To specify the alignment, I defined as follows at first:

```rust
pub struct ucontext_t {
    ...
    #[repr(align(8))]
    uc_regspace: [::c_ulong; 128],
}
```

(from https://github.com/rust-lang/libc/pull/1622/commits/546a690047f01e85dd50f5b0967fa03be5294a3b#diff-8c71e51ad9afb09c731c4ec6ac99ed44R45)

But this is invalid as the style checker warns.

So I [defined the separate type for `uc_regspace`](https://github.com/rust-lang/libc/pull/1622/commits/37d06c76a55c4b723dd2c1fb4364cb87de8f7b2b), but that was rejected by the test because the separate type doesn't exist in C, and I didn't know the way to ignore in `libc-test/build.rs`.
But now you taught me the how, now I have several questions:

- (1) In the first place, are this line (`__align: [::c_longlong; 0]`) and [this commit](https://github.com/rust-lang/libc/pull/1622/commits/37d06c76a55c4b723dd2c1fb4364cb87de8f7b2b) the right way to set alignment for `uc_regspace: [::c_ulong; 128]`? I'm new to memory alignment...
- (2) Should I recreate the new separate type for `uc_regspace` in a `if #[cfg(align)]` block?
- (3) If yes, should I support for older Rusts like `src/unix/uclibc/x86_64/align.rs` and `src/unix/uclibc/x86_64/no_align.rs`?

Reply:
@gnzlbg ping.

Reply:
> In the first place, are this line (__align: [::c_longlong; 0]) and this commit the right way to set alignment for uc_regspace: [::c_ulong; 128]? I'm new to memory alignment...

I think that's the simplest way to achieve that, simpler than by defining a new type, so if that works, I'd prefer this over defining a different type with the `repr(align(N))` attribute, since it avoids the need for workarounds.

Reply:
> I think that's the simplest way to achieve that, simpler than by defining a new type

Um... 😕  I asked you because the first way (attaching `#[repr(align(8))]` to the field of `struct`) doesn't seem to work, as the style checker warns that `#[repr(align(8))]` is unused.

Reply:
Ah, okay, sorry I got it wrong. So I leave this code as is.